client/src/endpoint/account.rs

```diff
@@ -617,3 +636,175 @@ mod ledger_operations_tests {
         );
     }
 }
+
+/// Represents the payments for account endpoint on the stellar horizon server.
+/// The endpoint will return all the Payment operations where a specific account
+/// is either the sender or receiver.
+///
+/// <https://www.stellar.org/developers/horizon/reference/endpoints/payments-for-account.html>
+///
+/// ## Example
+/// ```
+/// # extern crate stellar_client;
+/// # extern crate stellar_resources;
+/// use stellar_client::sync::Client;
+/// use stellar_client::endpoint::{account, payment};
+/// use stellar_resources::operation::OperationKind;
+///
+/// # fn main() {
+/// let client = Client::horizon_test().unwrap();
+///
+/// // Grab payments and associated account to ensure an account with payments
+/// let all_payments = client.request(payment::All::default().limit(1)).unwrap();
+/// let payment      = &all_payments.records()[0];
+/// let account_id   = match payment.kind() {
```

Comment:
Getting this `match` block right turned into a nice Rust learning exercise. 

@kbacha can you give this one more thumbs up for appropriate usage before I merge?